instance/tasks.py

```diff
@@ -76,47 +74,81 @@ def start_deployment(
     logger.info('Retrieving deployment: ID=%s', instance_ref_id)
     deployment = OpenEdXDeployment.objects.get(pk=deployment_id)
 
+    # For redundancy - if app server deployment was successful even after cancelling, we prevent the change in
+    # active appserver here
     if deployment.cancelled:
         logger.info('Deployment %s was cancelled, returning.', deployment.id)
-        return False
+        return
 
     logger.info('Spawning servers for deployment %s [%s]', deployment, deployment.id)
+    old_server_ids = list(instance.appserver_set.filter(_is_active=True).values_list('id', flat=True))
     # Launch configured number of appservers for instance
-    appserver_spawn_tasks = spawn_appserver.map(
-        (instance_ref_id, mark_active_on_success, False, num_attempts, success_tag, failure_tag, deployment.id)
-        for _ in range(instance.openedx_appserver_count)
-    )
+    for _ in range(instance.openedx_appserver_count):
+        # NOTE: Though this does provision multiple servers in parallel, issues have been known to occur
+        # for deploying multiple instances at once in the case of outstanding migrations.
+        spawn_appserver(
+            instance_ref_id=instance_ref_id,
+            mark_active_on_success=mark_active_on_success,
+            num_attempts=num_attempts,
+            success_tag=success_tag,
+            failure_tag=failure_tag,
+            deployment_id=deployment.id,
+            old_server_ids=old_server_ids,
+            target_count=instance.openedx_appserver_count,
+        )
+
 
-    appserver_ids = appserver_spawn_tasks.get(blocking=True)
+@db_task()
+def check_deactivation(
+        result: bool,
+        instance_ref_id: int = None,
+        deployment_id: int = None,
+        old_server_ids: List[int] = None,
```

Comment:
See [this comment](https://github.com/open-craft/opencraft/pull/628/files#r470525805)